Sources/Brave/Frontend/Browser/BrowserViewController/BrowserViewController+Wallet.swift

```diff
@@ -409,55 +409,65 @@ extension Tab: BraveWalletEventsListener {
     emitEthereumEvent(.ethereumAccountsChanged(accounts: []))
     emitEthereumEvent(.ethereumAccountsChanged(accounts: accounts))
-    updateEthereumProperties()
+    
+    Task {
+      await updateEthereumProperties()
+    }
   }
   
-  func updateEthereumProperties() {
+  @MainActor
+  func updateEthereumProperties() async {
     guard !isPrivate,
           let keyringService = BraveWallet.KeyringServiceFactory.get(privateMode: false),
           let walletService = BraveWallet.ServiceFactory.get(privateMode: false),
           Preferences.Wallet.defaultEthWallet.value == Preferences.Wallet.WalletType.brave.rawValue else {
       return
     }
-    Task { @MainActor in
-      // Turn an optional value into a string (or quoted string in case of the value being a string) or
-      // return `undefined`
-      func valueOrUndefined<T>(_ value: T?) -> String {
-        switch value {
-        case .some(let string as String):
-          return "\"\(string)\""
-        case .some(let value):
-          return "\(value)"
-        case .none:
-          return "undefined"
-        }
+    
+    // Turn an optional value into a string (or quoted string in case of the value being a string) or
+    // return `undefined`
+    func valueOrUndefined<T>(_ value: T?) -> String {
+      switch value {
+      case .some(let string as String):
+        return "\"\(string)\""
+      case .some(let value):
+        return "\(value)"
+      case .none:
+        return "undefined"
       }
-      guard let webView = webView, let provider = walletEthProvider else {
-        return
+    }
+    guard let webView = webView, let provider = walletEthProvider else {
+      return
+    }
+    
+    let chainId = await provider.chainId()
+    await withTaskGroup(of: Void.self) { group in
+      group.addTask { @MainActor in
+        await webView.evaluateSafeJavaScript(
+          functionName: "window.ethereum.chainId = \"\(chainId)\"",
+          contentWorld: EthereumProviderScriptHandler.scriptSandbox,
+          asFunction: false
+        )
+      }
+      
+      group.addTask { @MainActor in
+        let networkVersion = valueOrUndefined(Int(chainId.removingHexPrefix, radix: 16))
+        await webView.evaluateSafeJavaScript(
+          functionName: "window.ethereum.networkVersion = \(networkVersion)",
+          contentWorld: EthereumProviderScriptHandler.scriptSandbox,
+          asFunction: false
+        )
+      }
+      
+      group.addTask { @MainActor in
+        let coin = await walletService.selectedCoin()
+        let accounts = await keyringService.keyringInfo(coin.keyringId).accountInfos.map(\.address)
+        let selectedAccount = valueOrUndefined(await self.allowedAccounts(coin, accounts: accounts).1.first)
+        await webView.evaluateSafeJavaScript(
+          functionName: "window.ethereum.selectedAddress = \(selectedAccount)",
+          contentWorld: EthereumProviderScriptHandler.scriptSandbox,
+          asFunction: false)
```

Comment:
I don't see how this change is necessary: 

1. You don't need a task group for this, and there's no reason to run them concurrently here at all if they're all main actor isolated (only 1 will run at a time anyways)
2. `evaluateSafeJavaScript` is already MainActor isolated, meaning the first 2 tasks don't need MainActor isolation on the task itself
3. The previous code is basically identical (calling `evaluateSafeJavaScript` 3 times without waiting for completion)

Reply:
I guess the 1 main difference is this version waits until they all finish technically, if you want this behavior I recommend just awaiting on them all 1-by-1, it would be the same as it currently is

Reply:
Fixed.